pkg/utils/utils.go

```diff
@@ -558,35 +566,25 @@ func vfIsReady(pciAddr string) (netlink.Link, error) {
 	return vfLink, nil
 }
 
-func setVfsAdminMac(iface *sriovnetworkv1.InterfaceExt) error {
-	glog.Infof("setVfsAdminMac(): device %s", iface.PciAddress)
-	pfLink, err := netlink.LinkByName(iface.Name)
+func setVfsAdminMac(vfAddr string, pfLink netlink.Link, isRdma bool) error {
+	glog.Infof("setVfsAdminMac(): VF %s", vfAddr)
+
+	vfID, err := dputils.GetVFID(vfAddr)
 	if err != nil {
-		glog.Errorf("setVfsAdminMac(): unable to get PF link for device %+v %q", iface, err)
+		glog.Errorf("setVfsAdminMac(): unable to get VF id %+v %q", vfAddr, err)
 		return err
 	}
-	vfs, err := dputils.GetVFList(iface.PciAddress)
+	vfLink, err := vfIsReady(vfAddr)
 	if err != nil {
+		glog.Errorf("setVfsAdminMac(): VF link is not ready for device %+v %q", vfAddr, err)
 		return err
 	}
-	for _, addr := range vfs {
-		vfID, err := dputils.GetVFID(addr)
-		if err != nil {
-			glog.Errorf("setVfsAdminMac(): unable to get VF id %+v %q", iface.PciAddress, err)
-			return err
-		}
-		vfLink, err := vfIsReady(addr)
-		if err != nil {
-			glog.Errorf("setVfsAdminMac(): VF link is not ready for device %+v %q", addr, err)
-			return err
-		}
-		if err := netlink.LinkSetVfHardwareAddr(pfLink, vfID, vfLink.Attrs().HardwareAddr); err != nil {
-			return err
-		}
-		if err = Unbind(addr); err != nil {
-			return err
-		}
-		if err = BindDefaultDriver(addr); err != nil {
+	if err := netlink.LinkSetVfHardwareAddr(pfLink, vfID, vfLink.Attrs().HardwareAddr); err != nil {
+		return err
+	}
+	if isRdma {
+		glog.Infof("setVfsAdminMac(): unbind driver for %s", vfAddr)
```

Comment:
should we add a method to handle the driver unbind and move the unbind logic out of SetVfAdminMac and SetVfGuid ?
so that these functions do exactly what their name say ? 

e.g add a function : `UnbindDriverIfNeeded()` and call it from `configureSriovDevice` ?

Reply:
fixed